quadcontrol/config.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Flight scenario parser
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include "config.h"
+#include <parser.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+
+#define SCENARIO_TYPES_NB   7
+#define SCENARIO_MAX_FIELDS 3
+
+#define BASE 10
+
+
+enum { config_scenarioID = 0, config_pidID, config_end };
+
+
+static struct {
+	void *result;
+	size_t sz;  /* number of elements in `result` */
+	int invCnt; /* number of converter invocations */
+} res[config_end];
+
+
+static int config_takeoffParse(const hmap_t *h)
+{
+	char *altStr, *timeStr, *endptr;
+	flight_mode_t *result = (flight_mode_t *)res[config_scenarioID].result;
```

Comment:
You always point to the first element of the result. 
```suggestion
       unsigned int id = res[config_scenarioID].invCnt;
	flight_mode_t *mode = (flight_mode_t *)res[config_scenarioID].result[id];
```